pkg/infradb/infradb.go

```diff
@@ -975,24 +976,25 @@ func UpdateVrfStatus(name string, resourceVersion string, notificationID string,
 
 			// Delete VNI from the VPN map
 			vpns := make(map[uint32]bool)
-			found, err = infradb.client.Get("vpns", &vpns)
-			if err != nil {
-				log.Println(err)
-				return err
-			}
-			if !found {
-				log.Println("UpdateVrfStatus(): No VPNs have been found")
-				return ErrKeyNotFound
-			}
 			if vrf.Spec.Vni != nil {
+				found, err = infradb.client.Get("vpns", &vpns)
+				if err != nil {
+					log.Println(err)
+					return err
+				}
+				if !found {
+					log.Println("UpdateVrfStatus(): No VPNs have been found")
+					return ErrKeyNotFound
+				}
+
 				delete(vpns, *vrf.Spec.Vni)
-			}
-			err = infradb.client.Set("vpns", &vpns)
-			if err != nil {
-				log.Println(err)
-				return err
-			}
 
+				err = infradb.client.Set("vpns", &vpns)
+				if err != nil {
+					log.Println(err)
+					return err
+				}
```

Comment:
This chunk looks very similar to the one above...
Can we create a dedicated function to do that and eliminate the duplication? Maybe `removeVniFromVpns`?

Reply:
Done